### DISCUSSION TITLE
How to set the value of map to inline.

### DISCUSSION THREAD

**Discussion by mondy** [pelletier/go-toml#926] [Questions]
I am trying to create a configuration file like cargo.toml.
I want to output the following. But I don't know how to use inline tag.
The key part is indefinite. The value part can be a string representing the version or a structure.
I want only the value part to be inline.

```toml
[dependencies]
bar = '2.0.0'
foo = {version = '1.0.0', registory = 'test'}
```

I executed the following code, but it does not work.

```go
package main

import (
	"fmt"
	"log"

	"github.com/pelletier/go-toml/v2"
)

type Root struct {
	Dependencies map[string]any `toml:"dependencies,inline"` // How to describe it.
}

type Dependency struct {
	Version  string `toml:"version"`
	Registry string `toml:"registory"`
}

func main() {
	root := Root{Dependencies: map[string]any{
		"bar": "2.0.0",
		"foo": Dependency{Version: "1.0.0", Registry: "test"},
	}}

	if data, err := toml.Marshal(root); err != nil {
		log.Fatalln(err)
	} else {
		fmt.Println(string(data))
	}
}
```

Output:
```toml
dependencies = {bar = '2.0.0', foo = {version = '1.0.0', registory = 'test'}}
```

Where should I put inline? Is this module feasible?

**Comment by mondy** [comment]
ping @pelletier

**Comment by pelletier** [comment]
Since the inline tag can only be applied to struct fields, either the keys need to be known at compile time, or a struct type needs to be generated dynamically.

**Reply by mondy** [reply]
Thank you for your reply.
I able to achieve this with the reflect package.
However, it would be more convenient if it could be realized with struct tags.
